tests/unit/assessment/test_pipelines.py

```diff
@@ -6,53 +6,51 @@
 from databricks.labs.ucx.assessment.pipelines import PipelineInfo, PipelinesCrawler
 
 from ..framework.mocks import MockBackend
+from . import workspace_client_mock
 
 
-def test_pipeline_assessment_with_config(mocker):
+def test_pipeline_assessment_with_config():
     sample_pipelines = [
         PipelineStateInfo(
             cluster_id=None,
             creator_user_name="abcde.defgh@databricks.com",
             latest_updates=None,
             name="New DLT Pipeline",
-            pipeline_id="0112eae7-9d11-4b40-a2b8-6c83cb3c7407",
+            pipeline_id="pipeline_id_with_conf",
             run_as_user_name="abcde.defgh@databricks.com",
             state=PipelineState.IDLE,
         )
     ]
 
-    ws = Mock()
-    config_dict = {
-        "spark.hadoop.fs.azure.account.auth.type.abcde.dfs.core.windows.net": "SAS",
-        "spark.hadoop.fs.azure.sas.token.provider.type.abcde.dfs."
-        "core.windows.net": "org.apache.hadoop.fs.azurebfs.sas.FixedSASTokenProvider",
-        "spark.hadoop.fs.azure.sas.fixed.token.abcde.dfs.core.windows.net": "{{secrets/abcde_access/sasFixedToken}}",
-    }
-    ws.pipelines.get().spec.configuration = config_dict
+    ws = workspace_client_mock(clusters="job-source-cluster.json")
```

Comment:
I've extended `workspace_client_mock` in https://github.com/databrickslabs/ucx/pull/923 to read pipeline spec from a json file in under `tests/unit/assessment/pipelines`, so you can rebase your code off of that.

Reply:
Rebased the code to use the change